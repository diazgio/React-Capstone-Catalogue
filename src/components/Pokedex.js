import React from 'react';

const Pokedex = () => {
  return (
    <div>
      this is my pokedex
    </div>
  )
}

export default Pokedex;
