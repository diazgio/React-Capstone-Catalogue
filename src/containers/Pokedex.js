import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import {
  Card, CardMedia, CardContent, Typography, TextField,
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import SearchIcon from '@material-ui/icons/Search';
import { fetchPokemonAsync, fetchNextPagePokemon } from '../actions/getData';
import Pagination from '../components/Pagination';
import toFirstCharUppercase from '../constant/constant';

let url = 'https://pokeapi.co/api/v2/pokemon?limit=24';

const useStyles = makeStyles(() => ({
  cardMedia: {
    margin: 'auto',
  },
  cardContent: {
    textAlign: 'center',
  },
}));

const Pokedex = () => {
  const classes = useStyles();
  const page = useSelector((state) => state.page);
  const pokemon = useSelector((state) => state.pokemon);
  const dispatch = useDispatch();
  const [filter, setFilter] = useState('');

  const handleSearchChange = (e) => {
    setFilter(e.target.value);
  };

  const nextPage = () => {
    url = page.next;
    dispatch(fetchPokemonAsync(url));
  };
  const previousPage = () => {
    if (page.previous) {
      url = page.previous;
      dispatch(fetchPokemonAsync(url));
    }
  };

  useEffect(() => {
    dispatch(fetchNextPagePokemon(url));
  }, [url]);

  useEffect(() => {
    dispatch(fetchPokemonAsync(url));
  }, [url]);

  return (
    <div className="pokedex-container" id="top">
      <div className="SearchBar">
        <div className="search-cont">
          <SearchIcon className="searchicon" />
          <TextField
            className="input-search"
            label="Pokemon"
            onChange={handleSearchChange}
          />
        </div>
        <Pagination
          nextPage={nextPage || null}
          previousPage={previousPage || null}
        />
      </div>
      <div className="pokecontainer">
        {pokemon ? pokemon.map((p) => (
          p.name.includes(filter) && (
            <div className="pokecard" key={p.name}>
              <Card className="poke-data">
                <CardMedia
                  className={classes.cardMedia}
                  image={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${p.id}.png`}
                  style={{ width: '130px', height: '130px' }}
                />
                <CardContent className={classes.cardContent}>
                  <Typography>
                    {p.id}
                    .
                    {' '}
                    {toFirstCharUppercase(p.name)}
                  </Typography>
                </CardContent>
                <Link className="info-link" to={`${p.id}`}>Info</Link>
              </Card>
            </div>
          )
        )) : <p>Hi</p>}
      </div>
      <a href="#top" className="topbtn">&#8593;</a>
    </div>
  );
};

export default Pokedex;
