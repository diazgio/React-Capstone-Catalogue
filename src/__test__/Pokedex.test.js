import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Pokedex from '../containers/Pokedex';

describe('Creates a component', () => {
  const middlewares = [thunk];
  const initialState = [];
  const mockStore = configureStore(middlewares);
  let store;
  it('renders correctly', () => {
    store = mockStore(initialState);
    const tree = renderer.create(
      <Provider store={store}>
        <Pokedex />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders a component', () => {
    store = mockStore(initialState);
    render(<Provider store={store}><Pokedex /></Provider>);
    const element = screen.getByText('Pokemon');
    expect(element).toBeInTheDocument();
  });
});
