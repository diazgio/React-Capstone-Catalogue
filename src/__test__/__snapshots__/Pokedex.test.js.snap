// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a component renders correctly 1`] = `
<div
  className="pokedex-container"
>
  <div
    className="SearchBar"
  >
    <div
      className="search-cont"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root searchicon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <div
        className="MuiFormControl-root MuiTextField-root input-search"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
        >
          Pokemon
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="Pagination-container"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Previous
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        Next
      </button>
    </div>
  </div>
  <div
    className="pokecontainer"
  >
    <p>
      Hi
    </p>
  </div>
</div>
`;
